File: typescript/examples/free-service-smoke.ts
import { awaitReceipt, client, emit, fixedTarget } from "./_shared";

async function main(): Promise<void> {
  const synapse = client();
  const target = await fixedTarget(synapse);
  const result = await synapse.invoke(target.serviceId, target.payload, {
    costUsdc: target.costUsdc,
    idempotencyKey: `typescript-free-smoke-${Date.now()}`,
  });
  const receipt = await awaitReceipt(synapse, result.invocationId);
  emit({
    language: "typescript",
    scenario: "free-service-smoke",
    serviceId: target.serviceId,
    invocationId: result.invocationId,
    status: result.status,
    chargedUsdc: String(receipt.chargedUsdc),
    receiptStatus: receipt.status,
  });
}

void main();

File: java/examples/src/main/java/ai/synapsenetwork/sdk/examples/FreeServiceSmoke.java
package ai.synapsenetwork.sdk.examples;

import ai.synapsenetwork.sdk.SynapseClient;

public final class FreeServiceSmoke {
  private FreeServiceSmoke() {}

  public static void main(String[] args) {
    SynapseClient client = ExampleSupport.client();
    var target = ExampleSupport.fixedTarget(client);
    SynapseClient.InvokeOptions options = new SynapseClient.InvokeOptions();
    options.costUsdc = target.costUsdc();
    options.idempotencyKey = "java-free-smoke-" + System.currentTimeMillis();
    var result = client.invoke(target.serviceId(), target.payload(), options);
    var receipt = ExampleSupport.awaitReceipt(client, result.invocationId());
    ExampleSupport.emit(
        "free-service-smoke",
        result.status(),
        result.invocationId(),
        receipt.chargedUsdc(),
        receipt.status(),
        target.serviceId());
  }
}
